File: app/src/main/java/com/norvera/confession/utils/BindingAdapters.java
package com.norvera.confession.utils;

import android.icu.text.MessageFormat;
import android.os.Build;
import android.widget.TextView;

import com.norvera.confession.data.models.CommandmentEntry;

import androidx.databinding.BindingAdapter;

public class BindingAdapters {

    @BindingAdapter("commandmentText")
    public static void commandmentText(TextView textView, CommandmentEntry commandmentEntry) {
        if (commandmentEntry.commandment != null ) {
            if (commandmentEntry.id < 11)
                textView.setText(commandmentEntry.text);
            else
                textView.setText(commandmentEntry.commandment);
        }
    }


    @BindingAdapter("ordinalText")
    public static void ordinalText(TextView textView, Number number) {
        textView.setText(getNumberOrdinal(number));
    }

    private static String getNumberOrdinal(Number number) {
        String format;
        if (number == null) {
            return null;
        }
        if (number.intValue() < 11){
           format = "Sins against the {0,spellout,%spellout-ordinal} commandment";
        } else {
            format = "Sins against the Ministry";
        }


        //todo move to string resource

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            return MessageFormat.format(format, number);
        } else {
            return "com.ibm.icu.text.MessageFormat.format(format, number)";
        }
    }
}

File: app/src/main/java/com/norvera/confession/utils/BindingAdapters.kt
package com.norvera.confession.utils

import android.graphics.Color
import android.icu.text.MessageFormat
import android.os.Build
import android.text.SpannableString
import android.text.Spanned
import android.widget.TextView
import androidx.databinding.BindingAdapter
import com.norvera.confession.data.models.CommandmentEntry


@BindingAdapter("commandmentText")
fun commandmentText(textView: TextView, commandmentEntry: CommandmentEntry) {

    val string = SpannableString("Text with relative size span")
    string.setSpan(RelativeSizeColorSpan(Color.RED,1.5f), 10, 24, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)

    val text = "I saw a fox in the wood. The fox had red fur."

    val pattern = "fox".toRegex()

    val found = pattern.findAll(text)

    found.forEach { f ->
        val m = f.value
        val idx = f.range
        println("$m found at indexes: $idx")
    }

    if (commandmentEntry.id < 11)
        textView.text = commandmentEntry.text
    else
        textView.text = commandmentEntry.commandment
}


@BindingAdapter("ordinalText")
fun ordinalText(textView: TextView, number: Number) {
    textView.text = getNumberOrdinal(number)
}

private fun getNumberOrdinal(number: Number?): String? {
    if (number == null) return null

    var format = "Sins against the Ministry"
    if (number.toInt() < 11) {
        format = "Sins against the {0,spellout,%spellout-ordinal} commandment"
    }


    //todo move to string resource

    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
        MessageFormat.format(format, number)
    } else {
        "com.ibm.icu.text.MessageFormat.format(format, number)"
    }
}

